src/makim/core.py

```diff
@@ -380,29 +383,29 @@ def _render_env_inplace(
         scope_id = scope_options.index(scope)
 
         env = deepcopy(dict(os.environ))
-        variables: dict = {}
+        variables: dict[str, Any] = {}
 
         if scope_id >= SCOPE_GLOBAL:
             env_user = self.global_data.get('env', {})
             env_file = self._load_dotenv(self.global_data)
             _render_env_inplace(env_user, env_file, variables, env)
-            variables.update(self._load_scoped_vars('global', env=env))
+            variables.update(self._load_scoped_vars('global'))
 
         if scope_id >= SCOPE_GROUP:
             env_user = self.group_data.get('env', {})
             env_file = self._load_dotenv(self.group_data)
             _render_env_inplace(env_user, env_file, variables, env)
-            variables.update(self._load_scoped_vars('group', env=env))
+            variables.update(self._load_scoped_vars('group'))
 
         if scope_id == SCOPE_TARGET:
             env_user = self.task_data.get('env', {})
             env_file = self._load_dotenv(self.task_data)
             _render_env_inplace(env_user, env_file, variables, env)
-            variables.update(self._load_scoped_vars('task', env=env))
+            variables.update(self._load_scoped_vars('task'))
 
         return env, variables
 
-    def _load_scoped_vars(self, scope: str, env) -> dict:
+    def _load_scoped_vars(self, scope: str) -> Any:
```

Comment:
I would say that this should be a dict[str, Any]

```suggestion
    def _load_scoped_vars(self, scope: str) -> dict[str, Any]:
```

and in order to fix the issue with the `return`

I think it should be something like this: `return cast(Dict[str, Any], fix_dict_keys_recursively(variables))`

it is still not perfect .. but I will think some ways to improve `fix_dict_keys_recursively` .. but for now please do this change.